test/prototype/safetensors/test_safetensors_support.py

```diff
@@ -85,6 +86,45 @@ def test_safetensors(self, config, act_pre_scale=False):
         output = model(*example_inputs)
         assert torch.equal(output, ref_output)
 
+    @parametrize(
+        "config, act_pre_scale",
+        [
+            (Float8DynamicActivationFloat8WeightConfig(granularity=PerRow()), False),
+            (Int4WeightOnlyConfig(), False),
+            (Int4WeightOnlyConfig(), True),
+            (Int4WeightOnlyConfig(int4_packing_format="tile_packed_to_4d"), False),
+            (IntxWeightOnlyConfig(), False),
+            (Int8DynamicActivationIntxWeightConfig(), False),
+        ],
+    )
+    def test_safetensors_sharded(self, config, act_pre_scale=False):
+        model = torch.nn.Sequential(
+            torch.nn.Linear(128, 256, dtype=torch.bfloat16, device="cuda")
+        )
+        quantize_(model, config)
+        if act_pre_scale:
+            model[0].weight.act_pre_scale = torch.ones(
+                (1), dtype=torch.bfloat16, device="cuda"
+            )
+
+        with tempfile.NamedTemporaryFile() as f:
+            tensors_data_dict, metadata = flatten_tensor_state_dict(model.state_dict())
+            save_file(tensors_data_dict, f.name, metadata=metadata)
+            tensors_data_dict, metadata = load_data(file_path=f.name, device="cuda")
+
+            del tensors_data_dict[
```

Comment:
nit: we can also delete the `act_pre_scale` to make sure optional tensors are working as well